src/main/kotlin/xyz/crossward/service/GroupService.kt

```diff
@@ -0,0 +1,67 @@
+package xyz.crossward.service
+
+import org.springframework.stereotype.Service
+import xyz.crossward.entities.Group
+import xyz.crossward.exception.BadRequestException
+import xyz.crossward.exception.NoContentException
+import xyz.crossward.repository.GroupRepository
+import java.util.*
+
+@Service
+class GroupService(
+        val groupRepository: GroupRepository
+) {
+
+
+    fun findGroupById(id: String): Group {
+        return groupRepository.findById(id).unwrap()
+                ?: throw NoContentException("Could not find group with id $id")
+    }
+
+    fun findGroupByName(name: String): Group {
+        return groupRepository.findGroupByName(name.lowercase()).findFirst().unwrap()
+                ?: throw NoContentException("Could not find group with name $name")
+    }
+
+    /**
+     * Creates a Telegram group
+     *
+     * @param group group to create
+     * @return group saved to the database
+     */
+    fun createTelegramGroup(group: Group): Group {
+        if (groupRepository.existsById(group.id)) {
+            throw BadRequestException("A group with ID ${group.id} already exists")
+        }
+        // create a telegram group
+        val savedGroup = Group(
+                id = group.id,
+                name = group.name.lowercase()
+        )
+        groupRepository.save(savedGroup)
+        return savedGroup
+    }
+
+
+    /**
+     * Creates a Website group, using name as ID
+     *
+     * @param group group to create
+     * @return group saved to the database
+     */
+    fun createWebsiteGroup(group: Group): Group {
```

Comment:
Is there any difference between this and the other service function `createTelegramGroup()`? There may need to be two separate endpoints in the controller, but if the service call is the same we should just remove the duplicated code.

Reply:
The difference is that the website uses the group's name as it's ID while telegram uses that telegram-provided ID as the ID. I believe this is similar to how we create new users

Reply:
Ahhh ok I see the difference now, looks good.